src/translator/batch_translator.cpp

```diff
@@ -11,17 +11,30 @@ namespace bergamot {
 BatchTranslator::BatchTranslator(DeviceId const device,
                                  std::vector<Ptr<Vocab const>> &vocabs,
                                  Ptr<Options> options,
-                                 const void * model_memory)
-    : device_(device), options_(options), vocabs_(&vocabs), model_memory_(model_memory) {}
+                                 const void * model_memory,
+                                 const size_t model_memory_size,
+                                 const void * shortlist_memory,
+                                 const size_t shortlist_memory_size)
+    : device_(device), options_(options), vocabs_(&vocabs),
+    model_memory_(model_memory), model_memory_size_(model_memory_size),
+    shortlist_memory_(shortlist_memory), shortlist_memory_size_(shortlist_memory_size) {}
 
 void BatchTranslator::initialize() {
```

Comment:
I understand a separate initialize is what @kpu perhaps meant by 

> Jerin comes from the default-contructors-are-scary school 

This exists because at some point the objective I was directed at was to match marian-encoder and I had to offload model loading from files (which as you might already know is slow) into threads to change 48*x -> x when there were 48 threads. If this can use professional help, feel free. (This is my fault, I should have left a note).